2024-07-11 12:30:37.580575: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:30:37.580778: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:30:45,990 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:30:46,088 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:30:46,089 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:30:46,089 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:30:47,379 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:30:47,380 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,396 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.39666666666666667}
INFO flwr 2024-07-11 12:30:47,396 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:30:47,521 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,544 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:30:47,546 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,554 | server.py:131 | fit progress: (1, 0.39351904025996964, {'accuracy': 0.8033333333333333}, 0.15818490000000018)
DEBUG flwr 2024-07-11 12:30:47,555 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,576 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:30:47,576 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:30:47,576 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,598 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,611 | server.py:131 | fit progress: (2, 0.3921149132886247, {'accuracy': 0.81}, 0.21430119999999953)
DEBUG flwr 2024-07-11 12:30:47,611 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,629 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:30:47,630 | server.py:227 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,649 | server.py:241 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,660 | server.py:131 | fit progress: (3, 0.3921190252315085, {'accuracy': 0.81}, 0.26382939999999877)
DEBUG flwr 2024-07-11 12:30:47,661 | server.py:178 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,679 | server.py:192 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:30:47,680 | server.py:227 | fit_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,697 | server.py:241 | fit_round 4 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,708 | server.py:131 | fit progress: (4, 0.392119067480454, {'accuracy': 0.81}, 0.3120995000000004)
DEBUG flwr 2024-07-11 12:30:47,708 | server.py:178 | evaluate_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,726 | server.py:192 | evaluate_round 4 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:30:47,726 | server.py:227 | fit_round 5: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:30:47,746 | server.py:241 | fit_round 5 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,757 | server.py:131 | fit progress: (5, 0.3921190702547943, {'accuracy': 0.81}, 0.3607959999999988)
DEBUG flwr 2024-07-11 12:30:47,757 | server.py:178 | evaluate_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:30:47,779 | server.py:192 | evaluate_round 5 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:30:47,779 | server.py:227 | fit_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:30:47,809 | server.py:241 | fit_round 6 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,821 | server.py:131 | fit progress: (6, 0.3950968256003092, {'accuracy': 0.8033333333333333}, 0.42476920000000007)
DEBUG flwr 2024-07-11 12:30:47,821 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:30:47,845 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:30:47,845 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:30:47,874 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,886 | server.py:131 | fit progress: (7, 0.3936517844012914, {'accuracy': 0.8033333333333333}, 0.4893156999999988)
DEBUG flwr 2024-07-11 12:30:47,886 | server.py:178 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:30:47,908 | server.py:192 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:30:47,908 | server.py:227 | fit_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:30:47,936 | server.py:241 | fit_round 8 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:47,947 | server.py:131 | fit progress: (8, 0.3936441979649535, {'accuracy': 0.8033333333333333}, 0.5516199000000004)
DEBUG flwr 2024-07-11 12:30:47,948 | server.py:178 | evaluate_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:30:47,971 | server.py:192 | evaluate_round 8 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:30:47,971 | server.py:227 | fit_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:30:47,999 | server.py:241 | fit_round 9 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:48,012 | server.py:131 | fit progress: (9, 0.3936450344934874, {'accuracy': 0.8033333333333333}, 0.6157143999999999)
DEBUG flwr 2024-07-11 12:30:48,012 | server.py:178 | evaluate_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:30:48,036 | server.py:192 | evaluate_round 9 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:30:48,037 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:30:48,074 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:30:48,087 | server.py:131 | fit progress: (10, 0.39201307097564403, {'accuracy': 0.8066666666666666}, 0.6905707000000003)
DEBUG flwr 2024-07-11 12:30:48,087 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:30:48,118 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:30:48,118 | server.py:153 | FL finished in 0.7216398000000002
INFO flwr 2024-07-11 12:30:48,118 | app.py:225 | app_fit: losses_distributed [(1, 0.395399022847414), (2, 0.39334135688841343), (3, 0.3933422137051821), (4, 0.3933422565460205), (5, 0.40103903611501057), (6, 0.4040231953064601), (7, 0.40338775586514247), (8, 0.40338457907949177), (9, 0.4033848437524977), (10, 0.4065522675003324)]
INFO flwr 2024-07-11 12:30:48,119 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:30:48,119 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:30:48,119 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.39351904025996964), (2, 0.3921149132886247), (3, 0.3921190252315085), (4, 0.392119067480454), (5, 0.3921190702547943), (6, 0.3950968256003092), (7, 0.3936517844012914), (8, 0.3936441979649535), (9, 0.3936450344934874), (10, 0.39201307097564403)]
INFO flwr 2024-07-11 12:30:48,119 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.39666666666666667), (1, 0.8033333333333333), (2, 0.81), (3, 0.81), (4, 0.81), (5, 0.81), (6, 0.8033333333333333), (7, 0.8033333333333333), (8, 0.8033333333333333), (9, 0.8033333333333333), (10, 0.8066666666666666)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96784205,  0.9533537 ,  0.82493392,  0.33218081,  0.28024786,
         0.08408969,  0.01037991,  0.11141194]]), array([-0.07013303])]
[array([[-1.96784205,  0.9533537 ,  0.82493392,  0.33218081,  0.28024786,
         0.08408969,  0.01037991,  0.11141194]]), array([-0.07013303])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03077348,  0.98429968,  0.7606408 ,  0.34618367,  0.25483532,
         0.00990136,  0.00701369, -0.02139261]]), array([0.24090268])]
[array([[-2.03077348,  0.98429968,  0.7606408 ,  0.34618367,  0.25483532,
         0.00990136,  0.00701369, -0.02139261]]), array([0.24090268])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03035648,  0.98412559,  0.76054117,  0.3462342 ,  0.2550044 ,
         0.0099311 ,  0.00693106, -0.02136611]]), array([0.24041123])]
[array([[-2.03035648,  0.98412559,  0.76054117,  0.3462342 ,  0.2550044 ,
         0.0099311 ,  0.00693106, -0.02136611]]), array([0.24041123])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03033341,  0.98412031,  0.76054722,  0.34624171,  0.25500928,
         0.00994705,  0.00693164, -0.02136215]]), array([0.24034586])]
[array([[-2.03033341,  0.98412031,  0.76054722,  0.34624171,  0.25500928,
         0.00994705,  0.00693164, -0.02136215]]), array([0.24034586])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.0303324 ,  0.9841199 ,  0.76054724,  0.34624179,  0.25500931,
         0.00994739,  0.00693169, -0.02136197]]), array([0.24034409])]
[array([[-2.0303324 ,  0.9841199 ,  0.76054724,  0.34624179,  0.25500931,
         0.00994739,  0.00693169, -0.02136197]]), array([0.24034409])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94773021,  1.04780597,  0.81181769,  0.37070273,  0.22990275,
         0.05108088,  0.01367223, -0.06402975]]), array([0.02747358])]
[array([[-1.94773021,  1.04780597,  0.81181769,  0.37070273,  0.22990275,
         0.05108088,  0.01367223, -0.06402975]]), array([0.02747358])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.97480891,  1.01384774,  0.7791226 ,  0.38362274,  0.19157254,
         0.07022477, -0.02707863, -0.02822903]]), array([0.08048355])]
[array([[-1.97480891,  1.01384774,  0.7791226 ,  0.38362274,  0.19157254,
         0.07022477, -0.02707863, -0.02822903]]), array([0.08048355])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.97662428,  1.01500323,  0.78025291,  0.38397526,  0.19184605,
         0.07030653, -0.02671672, -0.02858592]]), array([0.07969198])]
[array([[-1.97662428,  1.01500323,  0.78025291,  0.38397526,  0.19184605,
         0.07030653, -0.02671672, -0.02858592]]), array([0.07969198])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.97645895,  1.01492212,  0.7801876 ,  0.38392596,  0.19180155,
         0.07029797, -0.02669419, -0.028597  ]]), array([0.07974767])]
[array([[-1.97645895,  1.01492212,  0.7801876 ,  0.38392596,  0.19180155,
         0.07029797, -0.02669419, -0.028597  ]]), array([0.07974767])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.99079097,  0.98021156,  0.80058046,  0.38233331,  0.17005754,
         0.1025759 , -0.02393239, -0.00717979]]), array([0.1174911])]
[array([[-1.99079097,  0.98021156,  0.80058046,  0.38233331,  0.17005754,
         0.1025759 , -0.02393239, -0.00717979]]), array([0.1174911])]
>>>>>>>>>>>>>run time:
0.359375s
